src/txn/txn.c

```diff
@@ -2789,7 +2790,8 @@ __wt_verbose_dump_txn_one(
 {
     WT_TXN *txn;
     WT_TXN_SHARED *txn_shared;
-    char buf[512];
+    uint32_t i;
+    char buf[512], snapshot_buf[512], snapshot_buf_tmp[32];
```

Comment:
These buffer sizes seem a bit small to me. 

The number of transaction IDs in a snapshot can be as large as the user configured `session_max`. There's no upper limit on this, but assuming the default `session_max` value of 100 the the longest possible string would be:
- 100 transaction IDs in the snapshot.
- Each transaction ID is 64 bits, so 20 characters long when printed in decimal.
- An additional 2 bytes for each `, ` between each reported transaction ID
For a maximum size of roughly 2,200 in `snapshot_buf`.

This is an extreme case, so 2048 bytes for `snapshot_buf` seems reasonable enough to catch almost every case. 
`buf` then needs to contain `snapshot_buf` along with all the other dumped data. Since `bud` was already 512 bytes can you update buf to 2,560 bytes?